cluster/gce/gci/configure-helper.sh

```diff
@@ -0,0 +1,684 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is for configuring kubernetes master and node instances. It is
+# uploaded in the manifests tar ball.
+
+set -o errexit
+set -o nounset
+
+function config-ip-firewall {
+  echo "Configuring IP firewall rules"
+  # The GCI image has host firewall which drop most inbound/forwarded packets.
+  # We need to add rules to accept all TCP/UDP packets.
+  if iptables -L INPUT | grep "Chain INPUT (policy DROP)" > /dev/null; then
+    echo "Add rules to accpet all inbound TCP/UDP packets"
+    iptables -A INPUT -w -p TCP -j ACCEPT
+    iptables -A INPUT -w -p UDP -j ACCEPT
+  fi
+  if iptables -L FORWARD | grep "Chain FORWARD (policy DROP)" > /dev/null; then
+    echo "Add rules to accpet all forwarded TCP/UDP packets"
+    iptables -A FORWARD -w -p TCP -j ACCEPT
+    iptables -A FORWARD -w -p UDP -j ACCEPT
+  fi
+}
+
+function create-dirs {
+  echo "Creating required directories"
+  mkdir -p /var/lib/kubelet
+  mkdir -p /etc/kubernetes/manifests
+  if [[ "${KUBERNETES_MASTER:-}" == "false" ]]; then
+    mkdir -p /var/lib/kube-proxy
+  fi
+}
+
+# Finds the master PD device; returns it in MASTER_PD_DEVICE
+function find-master-pd {
+  MASTER_PD_DEVICE=""
+  # TODO(zmerlynn): GKE is still lagging in master-pd creation
+  if [[ ! -e /dev/disk/by-id/google-master-pd ]]; then
+    return
+  fi
+  device_info=$(ls -l /dev/disk/by-id/google-master-pd)
+  relative_path=${device_info##* }
+  MASTER_PD_DEVICE="/dev/disk/by-id/${relative_path}"
+}
+
+# Mounts a persistent disk (formatting if needed) to store the persistent data
+# on the master -- etcd's data, a few settings, and security certs/keys/tokens.
+# safe_format_and_mount only formats an unformatted disk, and mkdir -p will
+# leave a directory be if it already exists.
+function mount-master-pd {
+  find-master-pd
+  if [[ -z "${MASTER_PD_DEVICE}" ]]; then
+    return
```

Comment:
When env var MASTER_PD_DEVICE is not set. I think it has a problem. If this var is unset, the script will exit due to "set -o nounset". We should check "${MASTER_PD_DEVICE:-}" instead. The ContainerVM's configure-vm.sh has the same problem. I will correct both.